pinecone/client.go

```diff
@@ -754,53 +763,99 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
 	return nil
 }
 
+// ConfigureIndexParams contains parameters for configuring an index. For both pod-based
+// and serverless indexes you can configure the DeletionProtection status for an index.
+// For pod-based indexes you can also configure the number of Replicas and the PodType.
+// Each of the fields are optional, but at least one field must be set.
```

Comment:
When you say "each of the fields" here, do you mean btwn `replicas` and `podtype` or btwn those 2 + also `deletion protection`?

Reply:
Between all 3, you can still pass `Replicas` or `PodType` to the server, it'll just respond with an error if it's a serverless index.

We enforce sending at least one argument in the struct on the client-side.